lib/mime.c

```diff
@@ -1778,6 +1778,23 @@ const char *Curl_mime_contenttype(const char *filename)
   return NULL;
 }
 
+static bool content_type_match(const char *contenttype, const char *target)
+{
+  size_t len = strlen(target);
+
+  if(contenttype && strncasecompare(contenttype, target, len))
+    switch(contenttype[len]) {
+    case '\0':
+    case '\t':
+    case '\r':
+    case '\n':
+    case ' ':
+    case ';':
```

Comment:
I'm not sure if HTML does not require it but there are some other escape sequences which might be relevant: https://en.wikipedia.org/wiki/Escape_sequences_in_C#Table_of_escape_sequences

Reply:
This is not for HTML, but for Content-Type header syntax: the check here just makes sure the "word" is finished. That means it rejects something like `multipart/form-data-fake`. Tested characters are the "legal ones" that could follow the content type (with some additional tolerance). I made it that way because it's much more direct than testing the _media/subtype_ syntax.